### ISSUE TITLE
State (data) handling

### ISSUE THREAD

**Issue by t-kramer** [CenterForTheBuiltEnvironment/comfort-dash#5]
- [x] look into save the website state in the URL. Here are some useful links [1](https://url.au.m.mimecastprotect.com/s/HWDUCP7LAXfvBgEmnUzfBu9V-6B?domain=community.plotly.com), [2](https://url.au.m.mimecastprotect.com/s/svCrCQnMBZfBnVjqLSPhJuJqpIJ?domain=gist.github.com), [3](https://url.au.m.mimecastprotect.com/s/vZa3CROND2unwYqlXsPizuAIf_n?domain=github.com), [4](https://url.au.m.mimecastprotect.com/s/A5d_CVARKgC0VBoK8UJsquGKtxZ?domain=community.plotly.com)
- [x] look at [Dash store](https://dash.plotly.com/dash-core-components/store)

**Comment by t-kramer** [comment]
@Echo11ovo @Ha02hen You can start doing some research on this.

**Comment by Echo11ovo** [comment]
I have completed the initial exploration, and now we have two potential approaches. As discussed in our previous meeting, we initially planned to divide pages based on functionality, using a URL structure like `localhost:8050/single/[model_name]/[parameter value]`.

However, during implementation, I discovered that this routing approach presents some issues within the Dash framework. Specifically, the page refreshes constantly whenever a user inputs parameters, significantly disrupting the user's input experience. The root of this problem likely stems from Dash's lack of optimization for this type of **dynamic URL structure** in its route management.

My understanding is that in the Dash framework, especially in multi-page applications, a URL structure like `/single/[model_name]/[parameter value]` might be interpreted as querying for specific files or resources under the "single" directory. Consequently, whenever parameters in the URL change, the framework perceives this as a new resource request, triggering a full page reload.

Given this issue, we may need to reconsider our URL structure design. A possible alternative could be to use query parameters, for example:

`localhost:8050/single?model=<model_name>&<parameter1>=<value1>&<parameter2>=<value2>`

This approach might reduce unnecessary page refreshes, as changes in query parameters are typically not viewed as new routes. Additionally, this structure aligns better with RESTful API design principles and may integrate more smoothly with Dash's callback system.

And here are my suggestions for setting up route parameters for the other functionalities:

- Compare functionality: a. Comparing different inputs for a single model: `/compare/inputs?model=<model_name>&set1=<params1>&set2=<params2>` b. Comparing multiple models with the same input: `/compare/models?models=<model1>,<model2>&<parameters>`
- Range functionality: `/range?model=<model_name>&<parameter_ranges>`

These URL structures provide a clear and intuitive way to represent different functionalities in your application, making it easier for users to understand and share specific views or comparisons.
